src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewViewModel.cs

```diff
@@ -0,0 +1,856 @@
+		{
+			var inner = GetVectorInnerImpl();
+			return inner.IndexOf(value);
+		}
+
+		private object[] GetMany(int startIndex)
+		{
+			var inner = GetVectorInnerImpl();
+			if (IsContentMode)
+			{
+				var vector = new List<object>();
+				int size = Size;
+				for (int i = 0; i < size; i++)
+				{
+					vector.Append(GetNodeAt(i).Content);
+				}
+				return vector.Skip(startIndex).ToArray();
+			}
+			return inner.Skip(startIndex).ToArray();
+		}
+
+		internal TreeViewNode GetNodeAt(int index)
+		{
+			var inner = GetVectorInnerImpl();
+			return (TreeViewNode)inner[index];
+		}
+
+		//void ViewModel.SetAt(uint32_t index, IInspectable const& value)
+		//{
+		//	var inner = GetVectorInnerImpl();
+		//	var current = inner.GetAt(index).as< TreeViewNode > ();
+		//	inner.SetAt(index, value);
+
+		//	TreeViewNode newNode = value.as< TreeViewNode > ();
+
+		//	var tvnCurrent = get_self<TreeViewNode>(current);
+		//	tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector[index]);
+		//	tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector[index]);
+
+		//	// Hook up events and replace tokens
+		//	var tvnNewNode = get_self<TreeViewNode>(newNode);
+		//	m_collectionChangedEventTokenVector[index] = tvnNewNode.ChildrenChanged({ this, &ViewModel.TreeViewNodeVectorChanged });
+		//	m_IsExpandedChangedEventTokenVector[index] = tvnNewNode.AddExpandedChanged({ this, &ViewModel.TreeViewNodePropertyChanged });
+		//}
+
+		//void ViewModel.InsertAt(uint32_t index, IInspectable const& value)
+		//{
+		//	GetVectorInnerImpl().InsertAt(index, value);
+		//	TreeViewNode newNode = value.as< TreeViewNode > ();
+
+		//	//Hook up events and save tokens
+		//	var tvnNewNode = get_self<TreeViewNode>(newNode);
+		//	m_collectionChangedEventTokenVector.insert(m_collectionChangedEventTokenVector.begin() + index, tvnNewNode.ChildrenChanged({ this, &ViewModel.TreeViewNodeVectorChanged }));
+		//	m_IsExpandedChangedEventTokenVector.insert(m_IsExpandedChangedEventTokenVector.begin() + index, tvnNewNode.AddExpandedChanged({ this, &ViewModel.TreeViewNodePropertyChanged }));
+		//}
+
+		//void ViewModel.RemoveAt(uint32_t index)
+		//{
+		//	var inner = GetVectorInnerImpl();
+		//	var current = inner.GetAt(index).as< TreeViewNode > ();
+		//	inner.RemoveAt(index);
+
+		//	// Unhook event handlers
+		//	var tvnCurrent = get_self<TreeViewNode>(current);
+		//	tvnCurrent.ChildrenChanged(m_collectionChangedEventTokenVector[index]);
+		//	tvnCurrent.RemoveExpandedChanged(m_IsExpandedChangedEventTokenVector[index]);
+
+		//	// Remove tokens from vectors
+		//	m_collectionChangedEventTokenVector.erase(m_collectionChangedEventTokenVector.begin() + index);
+		//	m_IsExpandedChangedEventTokenVector.erase(m_IsExpandedChangedEventTokenVector.begin() + index);
+		//}
+
+		private void Append(object value)
+		{
+			GetVectorInnerImpl().Add(value);
+			TreeViewNode newNode = (TreeViewNode)value;
+
+			// Hook up events and save tokens
+			var tvnNewNode = newNode;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'tvnNewNode'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)